src/ixdat/readers/cinfdata_db.py

```diff
@@ -0,0 +1,364 @@
+        self.group_meta = db.get_metadata_group(self.token)
+        self.meta = self.group_meta[list(self.group_meta.keys())[0]]
+
+        self.set_sample_name()
+        self.set_name()
+        self.set_tstamp()
+
+        spectrum_list = []
+        for i, key in enumerate(self.group_meta.keys()):
+            if self.group_meta[key]["type"] == 2:
+                self.x_name = "Binding energy / eV"
+                self.x_unit_name = "eV"
+                self.field_name = "Counts per second"
+                self.field_unit = "n/s"
+                self.technique = "XPS"
+                obj_as_dict = self.create_spectrum(key)  # group_data, group_meta, key)
+                obj_as_dict["name"] = self.group_meta[key]["name"]
+                spectrum_list.append(self.measurement_class.from_dict(obj_as_dict))
+
+            elif self.group_meta[key]["type"] == 4:
+                self.x_name = "Mass [AMU]"
+                self.x_unit_name = "m/z"
+                self.field_name = "Current"
+                self.field_unit = "[A]"
+                self.technique = "MS_spectrum"
+                obj_as_dict = self.create_spectrum(key)
+                obj_as_dict["name"] = self.sample_name
+                spectrum_list.append(self.measurement_class.from_dict(obj_as_dict))
+
+            else:
+                pass
+
+        if not spectrum_list:
+            warnings.warn(
+                f"No spectrum was found using '{self.token}' "
+                f" and group_column: '{self.grouping_column}'",
+                stacklevel=2,
+            )
+            return None
+        elif self.mass_scans:
+            return spectrum_list
+        elif len(spectrum_list) == 1:
+            return spectrum_list[0]
+        else:
+            try:
+                return SpectrumSeries.from_spectrum_list(spectrum_list)
+            except ValueError:
+                warnings.warn(
+                    "Could not return SpectrumSeries from list of spectrums "
+                    f"using '{self.token}' and group column: "
+                    f"'{self.grouping_column}'. \n"
+                    " Return list of all Spectrums.",
+                    stacklevel=2,
+                )
+
+                return spectrum_list
+
+    def create_spectrum(self, key):
+        x_col = self.group_data[key][:, 0]
+        y_col = self.group_data[key][:, 1]
+        tstamp = self.group_meta[key]["unixtime"]
+
+        xseries = DataSeries(data=x_col, name=self.x_name, unit_name=self.x_unit_name)
+        field = Field(
+            data=y_col,
+            name=self.field_name,
+            unit_name=self.field_unit,
+            axes_series=[
+                xseries,
+            ],
+        )
+
+        obj_as_dict = {
+            "sample_name": self.sample_name,
+            "technique": self.technique,
+            "field": field,
+            "tstamp": tstamp,
+        }
+
+        return obj_as_dict
```

Comment:
It might just as well. I think is from some of the initial thoughts on how to add relevant mass_spectras to a measurement from a specific timestamp.

but if the create spectrum method is to be removed this will also.